sdk/identity/src/oauth2_http_client.rs

```diff
@@ -0,0 +1,103 @@
+//! Implements the oauth2 crate http client interface using an `azure_core::HttpClient` instance.
+//! https://docs.rs/oauth2/latest/oauth2/#importing-oauth2-selecting-an-http-client-interface
+
+use azure_core::{
+    error::{Error, ErrorKind, ResultExt},
+    HttpClient, Request,
+};
+use std::{collections::HashMap, str::FromStr, sync::Arc};
+
+pub(crate) struct Oauth2HttpClient {
+    http_client: Arc<dyn HttpClient>,
+}
+
+impl Oauth2HttpClient {
+    /// Create a new Oauth2HttpClient
+    pub fn new(http_client: Arc<dyn HttpClient>) -> Self {
+        Self { http_client }
+    }
+
+    pub(crate) async fn request(
+        &self,
+        oauth2_request: oauth2::HttpRequest,
+    ) -> Result<oauth2::HttpResponse, azure_core::error::Error> {
+        let method = try_from_method(&oauth2_request.method)?;
+        let mut request = Request::new(oauth2_request.url, method);
+        for (name, value) in to_headers(&oauth2_request.headers) {
+            request.insert_header(name, value)
+        }
+        request.set_body(oauth2_request.body);
+        let response = self.http_client.execute_request(&request).await?;
+        let status_code = try_from_status(response.status())?;
+        let headers = try_from_headers(response.headers())?;
+        let body = response.into_body().collect().await?.to_vec();
+        Ok(oauth2::HttpResponse {
+            status_code,
+            headers,
+            body,
+        })
+    }
+}
+
+fn try_from_method(method: &oauth2::http::Method) -> azure_core::Result<azure_core::Method> {
```

Comment:
```suggestion
fn try_from_method(method: oauth2::http::Method) -> azure_core::Result<azure_core::Method> {
```

Reply:
Why?

Reply:
If `Method` is `Copy` then this is usually more efficient (not that performance matters at all here). Clippy has a warning for this.

Reply:
There is no clippy warning. `oauth2::http::Method` is not `Copy`:

``` rust
#[derive(Clone, PartialEq, Eq, Hash)]
pub struct Method(Inner);
```